api/functions/verification_email.py

```diff
@@ -0,0 +1,61 @@
+import os
+import time
+
+from flask import request
+from graphql import GraphQLError
+from requests import HTTPError
+
+from json_web_token import tokenize
+from notifications_python_client.notifications import NotificationsAPIClient
+
+
+def send_verification_email(user):
+    """
+    This function allows a user object to be passed in during account creation
+    and send an email to be used for verifying accounts
+    :param user: A instance of /models/User.py
+    :return: None
+    """
+    # Create Notify Client
+    notify_client = NotificationsAPIClient(
+        api_key=os.getenv("NOTIFICATION_API_KEY"),
+        base_url=os.getenv("NOTIFICATION_API_URL"),
+    )
```

Comment:
I think this is a dependency that should be passed in rather than instantiated here. This would make the `send_verification_email` function way easier to test.
Maybe something like `def send_verification_email(user, client = NotificationsAPIClient(...)):`. 
That way in the tests you coud pass in a [stub](https://github.com/alex/pretend#how-do-i-use-pretend) and assert that the `send_email_notification` method is called on it with the args you expected. [MagicMock](https://docs.python.org/dev/library/unittest.mock.html) is another way to create an object that will let you do this.
This make sure our tests aren't hitting the network and Notify when they run.

Reply:
So for testing the actually functionality we have been using our test key, and following along with the Notify [docs](https://docs.notifications.service.gov.uk/python.html#other-testing) in regards to testing.

Reply:
I looked and the thing I was talking about needs a few other changes. Let's get this in, and refactor it after.